pallets/alt-vr/src/lib.rs

```diff
@@ -0,0 +1,203 @@
+        #[pallet::constant]
+        type MaxNameLength: Get<u8>;
+    }
+
+    #[pallet::pallet]
+    #[pallet::generate_store(pub(super) trait Store)]
+    #[pallet::storage_version(STORAGE_VERSION)]
+    pub struct Pallet<T>(PhantomData<T>);
+
+    #[pallet::hooks]
+    impl<T: Config> Hooks<BlockNumberFor<T>> for Pallet<T> {}
+
+    #[pallet::call]
+    impl<T: Config> Pallet<T> {
+        /// Creates altvr informations.
+        #[pallet::weight(T::WeightInfo::create_altvr())]
+        #[transactional]
+        pub fn create_altvr(
+            origin: OriginFor<T>,
+            username: ternoa::String,
+            vchatname: ternoa::String,
+        ) -> DispatchResultWithPostInfo {
+            let owner = ensure_signed(origin)?;
+
+            let username_lower_bound = username.len() >= T::MinNameLength::get() as usize;
+            let username_upper_bound = username.len() <= T::MaxNameLength::get() as usize;
+            let vchatname_lower_bound = vchatname.len() >= T::MinNameLength::get() as usize;
+            let vchatname_upper_bound = vchatname.len() <= T::MaxNameLength::get() as usize;
+
+            ensure!(username_lower_bound, Error::<T>::TooShortUsername);
+            ensure!(username_upper_bound, Error::<T>::TooLongUsername);
+            ensure!(vchatname_lower_bound, Error::<T>::TooShortVchatname);
+            ensure!(vchatname_upper_bound, Error::<T>::TooLongVchatname);
+
+            // Create Altvr data
+            Self::new_altvr(&owner, &username, &vchatname);
+
+            let event = Event::AltvrCreated(owner, username, vchatname);
+            Self::deposit_event(event);
+
+            Ok(().into())
+        }
+
+        /// Update Altvr username of the caller
+        #[pallet::weight(T::WeightInfo::update_username())]
+        #[transactional]
+        pub fn update_username(
+            origin: OriginFor<T>,
+            username: ternoa::String,
+        ) -> DispatchResultWithPostInfo {
+            let owner = ensure_signed(origin)?;
+
+            let username_lower_bound = username.len() >= T::MinNameLength::get() as usize;
+            let username_upper_bound = username.len() <= T::MaxNameLength::get() as usize;
+
+            ensure!(username_lower_bound, Error::<T>::TooShortUsername);
+            ensure!(username_upper_bound, Error::<T>::TooLongUsername);
+
+            Altvrs::<T>::try_mutate(owner.clone(), |res| -> Result<(), Error<T>> {
+                let altvrdata = res.as_mut().ok_or(Error::<T>::UserNotFound)?;
+                altvrdata.username = username.clone();
+                Ok(())
+            })?;
+
+            let event = Event::AltvrUsernameUpdated(owner, username);
+            Self::deposit_event(event);
+
+            Ok(().into())
+        }
+
+        /// Update Altvr vchatname of the caller
+        #[pallet::weight(T::WeightInfo::update_vchatname())]
+        #[transactional]
+        pub fn update_vchatname(
```

Comment:
Instead of `update_vchatname` use `set_vchatname`.

Reply:
done